workflow/cron/operator.go

```diff
@@ -370,6 +378,17 @@ func (woc *cronWfOperationCtx) reportCronWorkflowError(conditionType v1alpha1.Co
 	}
 }
 
+func (woc *cronWfOperationCtx) annotateLastUsedSchedule() {
+	lastUsedSchedule, exists := woc.cronWf.Annotations[common.AnnotationKeyCronWfLastUsedSchedule]
+	// If last-used-schedule does not exist, or if it does not match the current schedule, update it
+	if !exists || lastUsedSchedule != woc.cronWf.Spec.GetScheduleString() {
+		if woc.cronWf.Annotations == nil {
+			woc.cronWf.Annotations = map[string]string{}
+		}
+		woc.cronWf.Annotations[common.AnnotationKeyCronWfLastUsedSchedule] = woc.cronWf.Spec.GetScheduleString()
```

Comment:
minor - I feel like we spend much time getting and setting annotations, what about if we had `cronWf.SetLastUsedSchedule()` (if that is even possible

Reply:
Done!